subsys/settings/src/settings_init.c

```diff
@@ -117,6 +117,53 @@ int settings_backend_init(void)
 
 	return rc;
 }
+
+#elif defined(CONFIG_SETTINGS_NVS)
+#include <device.h>
+#include <flash_map.h>
+#include "settings/settings_nvs.h"
+
+#define NVS_MAX_SECTOR_NUM (DT_FLASH_AREA_STORAGE_SIZE / \
+			    DT_FLASH_ERASE_BLOCK_SIZE)
+
+static struct settings_nvs default_settings_nvs;
+
+int settings_backend_init(void)
+{
+	int rc;
+	u32_t cnt = NVS_MAX_SECTOR_NUM;
+	struct flash_sector sectors[NVS_MAX_SECTOR_NUM];
+
+	rc = flash_area_get_sectors(DT_FLASH_AREA_STORAGE_ID, &cnt, sectors);
+	if (rc) {
+		return rc;
+	}
+	if (!cnt) {
+		return -ENOMEM;
+	}
+
+	/* define the nvs file system using the page_info */
+	default_settings_nvs.cf_nvs.sector_size = sectors[0].fs_size;
+	default_settings_nvs.cf_nvs.sector_count = cnt;
+	default_settings_nvs.cf_nvs.offset = DT_FLASH_AREA_STORAGE_OFFSET;
```

Comment:
```suggestion
	rc = flash_area_open(DT_FLASH_AREA_STORAGE_ID, &fa);
	if (rc) {
		zło;
	}
	default_settings_nvs.cf_nvs.offset = fa.fa_off;
	default_settings_nvs.flash_dev_name = flash_area_get_device(const struct flash_area *fa);
```

Reply:
It could be even simpler:

default_settings_nvs.flash_dev_name = fa.fa_dev_name

Reply:
default_settings_nvs.flash_dev_name = device_get_binding(fa.fa_dev_name);

Reply:
so actually exactly what is under the API.

Reply:
It's ok in the code.